src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1052,3 +1057,45 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
     // activate the ramping in scratchProcess()
     m_ramp[deck] = true;
 }
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const ControllerScriptInterfaceLegacy::WellKnownCharsets targetCharset,
+        const QString& value) {
+    switch (targetCharset) {
+    case WellKnownCharsets::US_ASCII:
+        return convertCharsetInternal(QStringLiteral("US-ASCII"), value);
+    case WellKnownCharsets::Latin1:
+    case WellKnownCharsets::ISO_8859_1:
+        return convertCharsetInternal(QStringLiteral("ISO-8859-1"), value);
+    case WellKnownCharsets::Latin9:
+    case WellKnownCharsets::ISO_8859_15:
+        return convertCharsetInternal(QStringLiteral("ISO-8859-15"), value);
+    case WellKnownCharsets::UCS2:
+    case WellKnownCharsets::ISO_10646_UCS_2:
+        return convertCharsetInternal(QStringLiteral("ISO-10646-UCS-2"), value);
+    case WellKnownCharsets::UTF_8:
+        return convertCharsetInternal(QStringLiteral("UTF-8"), value);
+    case WellKnownCharsets::UTF_16BE:
+        return convertCharsetInternal(QStringLiteral("UTF-16BE"), value);
+    case WellKnownCharsets::UTF_16LE:
+        return convertCharsetInternal(QStringLiteral("UTF-16LE"), value);
+    }
+    m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unknown charset specified"));
+    return QByteArray();
+}
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharsetInternal(
+        const QString& targetCharset, const QString& value) {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
+    QAnyStringView encoderName = QAnyStringView(targetCharset);
+#else
+    QByteArray encoderNameArray = targetCharset.toUtf8();
+    const char* encoderName = encoderNameArray.constData();
+#endif
+    QStringEncoder fromUtf16 = QStringEncoder(encoderName);
+    if (!fromUtf16.isValid()) {
```

Comment:
see the commit message, writing the replacement char is better than replacing with null bytes IMO.

Reply:
This may be a problem. During the tests I noticed that replacement char varies between Ubuntu and Fedora (maybe between Qt versions). Replace invalid chars with `\0x00` is the most predictable option.

Reply:
Are you sure? I think it depends on the codec. The Qt docs say they use [`QChar::ReplacementCharacter`](https://doc.qt.io/qt-6/qchar.html#SpecialCharacter-enum) [or a question mark](https://doc.qt.io/qt-6/qstringconverter.html#Flag-enum).

Reply:
[This is what I noticed here](https://github.com/mixxxdj/mixxx/pull/13772#issuecomment-2437566659). The relevant commit is probably [this one](https://github.com/mixxxdj/mixxx/commit/a12e5e3b1f0de67f0950dfca248da0d911132b2d).

Reply:
I'll try to reproduce later this week if I have time but I don't think this should be a blocker. We'll have time to change it later or introduce a new API if this proves unstable.

Reply:
@christophehenry The flags here are the last open point, did you find to check this out?

Reply:
Oh sorry I didn't know you were waiting for me. I couldn't reproduce this. Feel free to merge as is.

Reply:
Ok, than lets merge this!